src/Microsoft.AspNet.OData.Shared/DefaultODataAPIHandler.cs

```diff
@@ -0,0 +1,179 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="DefaultODataAPIHandler.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// This is the default ODataAPIHandler for CLR types. This class has default get, create and update
+    /// methods that are used to patch an original collection when the collection is provided.
+    /// </summary>
+    /// <typeparam name="TStructuralType"></typeparam>
+    internal class DefaultODataAPIHandler<TStructuralType> : ODataAPIHandler<TStructuralType> where TStructuralType :class
+    {
+        Type clrType;
+        ICollection<TStructuralType> originalList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultODataAPIHandler{TStructuralType}"/> class.
+        /// </summary>
+        /// <param name="originalList">Original collection of the type which needs to be updated.</param>
+        public DefaultODataAPIHandler(ICollection<TStructuralType> originalList)
+        {
+            this.clrType = typeof(TStructuralType);
+            this.originalList = originalList ?? new List<TStructuralType>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out TStructuralType originalObject, out string errorMessage)
+        {
+            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(TStructuralType);
+
+            try
+            {
```

Comment:
All the logic executed within this `try...catch` block is our own and it doesn't seem to me like any of the statement is likely to encounter an exceptional scenario that causes an exception to be thrown